web-ui/src/components/certifications/Certifications.jsx

```diff
@@ -0,0 +1,254 @@
+import PropTypes from 'prop-types';
+import React, { useCallback, useEffect, useState } from 'react';
+
+import {
+  Button,
+  Dialog,
+  DialogContent,
+  DialogTitle,
+  TextField
+} from '@mui/material';
+import Autocomplete, { createFilterOptions } from '@mui/material/Autocomplete';
+
+import ConfirmationDialog from '../dialogs/ConfirmationDialog';
+
+import './Certifications.css';
+
+const certificationBaseUrl = 'http://localhost:3000/certification';
+
+const propTypes = {
+  forceUpdate: PropTypes.func,
+  open: PropTypes.bool,
+  onClose: PropTypes.func
+};
+
+const Certifications = ({ forceUpdate = () => {}, open, onClose }) => {
+  const [adding, setAdding] = useState(true); // true to add, false to edit
+  const [badgeUrl, setBadgeUrl] = useState('');
+  const [certificationMap, setCertificationMap] = useState({});
+  const [certifications, setCertifications] = useState([]);
+  const [confirmDeleteOpen, setConfirmDeleteOpen] = useState(false);
+  const [mergeDialogOpen, setMergeDialogOpen] = useState(false);
+  const [name, setName] = useState('');
+  const [selectedCertification, setSelectedCertification] = useState(null);
+  const [selectedTarget, setSelectedTarget] = useState(null);
+
+  const close = () => {
+    reset();
+    onClose();
+  };
+
+  const reset = () => {
+    setAdding(true);
+    setBadgeUrl('');
+    setName('');
+    setSelectedCertification(null);
+    setSelectedTarget(null);
+  };
+
+  const loadCertifications = useCallback(async () => {
+    try {
+      let res = await fetch(certificationBaseUrl);
+      const certs = await res.json();
+      setCertifications(certs.sort((c1, c2) => c1.name.localeCompare(c2.name)));
+
+      const certMap = certs.reduce((map, cert) => {
+        map[cert.id] = cert;
+        return map;
+      }, {});
+      setCertificationMap(certMap);
+    } catch (err) {
+      console.error(err);
+    }
+  }, []);
+
+  useEffect(() => {
+    loadCertifications();
+  }, []);
+
+  const certificationSelect = useCallback(
```

Comment:
This is used to create two `AutoComplete` components, one for selecting a certification to be managed,
and another for selecting a certification into which the first will be merged. Merging is useful when we discover that two certifications were created that actually describe the same certification and we have already created earned certifications of both types.